### PR TITLE
mail-filter/spamassassin: add 4.0.0

### PR DESCRIPTION
**Pull request by Philippe23**
- Bump to 4.0.0
- Add new dependencies
  - dev-perl/Archive-Zip, dev-perl/Email-Address-XS, dev-perl/IO-String, dev-perl/Mail-DMARC, dev-perl/Net-LibIDN2
- Clean up old dependencies
  - virtual/perl-Archive-Tar, virtual/perl-Archive-Tar, virtual/perl-Time-HiRes, dev-perl/HTTP-Date, dev-perl/Net-CIDR-Lite
- Remove arm, arm64, hppa, ia64, ppc, ppc64, riscv, s390, & sparc KEYWORDS; will re-add/re-keyword shortly.

Bug: https://bugs.gentoo.org/886477
Closes: https://github.com/gentoo/gentoo/pull/28759
Signed-off-by: Philippe Chaintreuil <gentoo_bugs_peep@parallaxshift.com>

---
Diff against previous for easy reference: 
```diff
--- spamassassin-3.4.6-r2.ebuild        2022-12-19 10:18:40.181543790 -0500
+++ spamassassin-4.0.0.ebuild   2022-12-23 11:21:13.367263733 -0500
@@ -1,19 +1,19 @@
 # Copyright 1999-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2

-EAPI=7
+EAPI=8

 inherit perl-functions systemd toolchain-funcs

 MY_P="Mail-SpamAssassin-${PV//_/-}"
-S="${WORKDIR}/${MY_P}"
 DESCRIPTION="An extensible mail filter which can identify and tag spam"
 HOMEPAGE="https://spamassassin.apache.org/"
 SRC_URI="mirror://apache/spamassassin/source/${MY_P}.tar.bz2"
+S="${WORKDIR}/${MY_P}"

 LICENSE="Apache-2.0 GPL-2"
 SLOT="0"
-KEYWORDS="~alpha amd64 arm arm64 ~hppa ~ia64 ppc ppc64 ~riscv ~s390 sparc x86 ~amd64-linux ~x86-linux"
+KEYWORDS="~alpha ~amd64 ~x86 ~amd64-linux ~x86-linux"
 IUSE="berkdb cron ipv6 ldap mysql postgres qmail sqlite ssl test"
 RESTRICT="!test? ( test )"

@@ -26,10 +26,7 @@
        dev-perl/HTML-Parser
        dev-perl/Net-DNS
        dev-perl/NetAddr-IP
-       virtual/perl-Archive-Tar
        virtual/perl-Digest-SHA
-       virtual/perl-IO-Zlib
-       virtual/perl-Time-HiRes
        ssl? (
                dev-libs/openssl:0=
        )"
@@ -43,15 +40,18 @@
 # We still need the old Digest-SHA1 because razor2 has not been ported
 # to Digest-SHA.
 OPTDEPEND="app-crypt/gnupg
+       dev-perl/Archive-Zip
        dev-perl/BSD-Resource
        dev-perl/Digest-SHA1
+       dev-perl/Email-Address-XS
        dev-perl/Encode-Detect
        || ( dev-perl/GeoIP2 dev-perl/Geo-IP )
-       dev-perl/HTTP-Date
+       dev-perl/IO-String
        dev-perl/Mail-DKIM
+       dev-perl/Mail-DMARC
        dev-perl/Mail-SPF
        dev-perl/Net-Patricia
-       dev-perl/Net-CIDR-Lite
+       dev-perl/Net-LibIDN2
        dev-util/re2c
        || ( net-misc/wget[ssl] net-misc/curl[ssl] )
        virtual/perl-MIME-Base64
@@ -82,7 +82,6 @@

 PATCHES=(
        "${FILESDIR}/mention-geoip.cf-in-init.pre.patch"
-       "${FILESDIR}/3.4.6-configure-clang16.patch"
 )

 # There are a few renames and use-dependent ones in src_istall as well.
```

### CODE REVIEW

File: mail-filter/spamassassin/spamassassin-4.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,321 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (ConiKost) [inline mail-filter/spamassassin/spamassassin-4.0.0.ebuild:4]: Please bump to EAPI=8

Comment (Philippe23) [reply mail-filter/spamassassin/spamassassin-4.0.0.ebuild:4]: 👍 Will do!  Thanks for flagging it.

File: mail-filter/spamassassin/spamassassin-4.0.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+inherit perl-functions systemd toolchain-funcs
+
+MY_P="Mail-SpamAssassin-${PV//_/-}"
+S="${WORKDIR}/${MY_P}"
```
Comment (ConiKost) [inline mail-filter/spamassassin/spamassassin-4.0.0.ebuild:9]: ```S``` goes after ```SRC_URI```. See skel.ebuild.

File: mail-filter/spamassassin/spamassassin-4.0.0.ebuild (hunk 1 above)
Comment (Philippe23) [reply mail-filter/spamassassin/spamassassin-4.0.0.ebuild]: Done.

File: mail-filter/spamassassin/spamassassin-4.0.0.ebuild (hunk 2 above)
Comment (Philippe23) [reply mail-filter/spamassassin/spamassassin-4.0.0.ebuild:9]: Done.

Comment (scardracs) [reply mail-filter/spamassassin/spamassassin-4.0.0.ebuild:9]: You may also want to push update, because it is still here

Comment (Philippe23) [reply mail-filter/spamassassin/spamassassin-4.0.0.ebuild]: Yeah, needed another push.  Thanks.

File: mail-filter/spamassassin/spamassassin-4.0.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+
+MY_P="Mail-SpamAssassin-${PV//_/-}"
+DESCRIPTION="An extensible mail filter which can identify and tag spam"
+HOMEPAGE="https://spamassassin.apache.org/"
+SRC_URI="mirror://apache/spamassassin/source/${MY_P}.tar.bz2"
+S="${WORKDIR}/${MY_P}"
+
+LICENSE="Apache-2.0 GPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux"
```
Comment (Flowdalic) [inline mail-filter/spamassassin/spamassassin-4.0.0.ebuild:16]: As per https://qa-reports.gentoo.org/output/gentoo-ci/e6292b4200/output.html#mail-filter/spamassassin you probably want to drop the arm and ppc keywords for new and create a (re)keywording bug for spamassasin 4.0.

File: mail-filter/spamassassin/spamassassin-4.0.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+	# We also set the path to the perl executable explictly. This will be
+	# used to create the initial shebang line in the scripts (bug 62276).
+	perl Makefile.PL \
+		PREFIX="${EPREFIX}/usr" \
+		INSTALLDIRS=vendor \
+		SYSCONFDIR="${EPREFIX}/etc" \
+		DATADIR="${EPREFIX}/usr/share/spamassassin" \
+		PERL_BIN="${EPREFIX}/usr/bin/perl" \
+		ENABLE_SSL="$(usex ssl)" \
+		DESTDIR="${D}" \
```
Comment (Flowdalic) [inline mail-filter/spamassassin/spamassassin-4.0.0.ebuild:139]: This is not new to the 4.0 ebuild, but using `D` in `src_configure` is, strictly speaking, invalid. Maybe this is no longer necessary with spamassassin 4.0?

Comment (thesamesam) [reply mail-filter/spamassassin/spamassassin-4.0.0.ebuild:139]: Unfortunately this is some Perl-related nonsense which we need to look into further (it's an issue with all Perl modules using a certain build system and even the eclass violates it), so I wouldn't block on it.

File: mail-filter/spamassassin/spamassassin-4.0.0.ebuild (hunk 3 above)
Comment (Philippe23) [reply mail-filter/spamassassin/spamassassin-4.0.0.ebuild]: Thanks for the guidance.  

Done.

File: mail-filter/spamassassin/spamassassin-4.0.0.ebuild (hunk 4 above)
Comment (Philippe23) [reply mail-filter/spamassassin/spamassassin-4.0.0.ebuild:139]: Yeah, unfortunately, what Sam said.

**Comment by Philippe23** [comment]
I didn't realize that `pkgcheck` would only output the first two arches.  Just did some iteration and the removed keyword arches list has grown.  Updated the comment at the top with the latest list.

**Pull request closed without merging**